### ISSUE TITLE
Kubectl e2e tests leave resources running after stop during upgrade

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#15112]
The following e2e tests are failing in a similar way during upgrades from 1.0 to HEAD:

- Kubernetes e2e suite.Kubectl client Guestbook application should create and stop a working application
- Kubernetes e2e suite.Kubectl client Kubectl label should update the label on a resource
- Kubernetes e2e suite.Kubectl client Simple pod should support exec
- Kubernetes e2e suite.Kubectl client Simple pod should support port-forward
- Kubernetes e2e suite.Kubectl client Update Demo should create and stop a replication controller

They are all leaving resources running after stop, and they all occur both (1) after the master has been upgraded, but before the nodes have been, *and* (2) after both the master and nodes have been upgraded.

See runs 33 - 39 of the kubernetes-upgrade-gke Jenkins job.

Let me know if you have any questions, or if there's anything I can do to help.

(This is part of #14778.)

**Comment by lavalamp** [comment]
can you link to some jenkins output? This doesn't seem specifically CSI so i changed the team.

**Comment by ihmccreery** [comment]
This is likely caused by #15123.

**Comment by ihmccreery** [comment]
Okay, update: the test was actually running against 1.2.0-alpha.xxx, so this is affecting HEAD, not 1.1.  Sorry about that.

For output, see http://kubekins.dls.corp.google.com/job/kubernetes-upgrade-gke-step3-e2e-old/1565/consoleFull.

```
00:40:25 Kubectl client Guestbook application 
00:40:25   should create and stop a working application
00:40:25   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:138
00:40:25 [BeforeEach] Kubectl client
00:40:25   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:70
00:40:25 >>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/.kube/config
00:40:30 INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-kubectl-xcdto
00:40:30 INFO: Get service account default in ns e2e-tests-kubectl-xcdto failed, ignoring for 5s: serviceaccounts "default" not found
00:40:35 INFO: Service account default in ns e2e-tests-kubectl-xcdto with secrets found. (5.004395766s)
00:40:35 [BeforeEach] Guestbook application
00:40:35   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:128
00:40:35 [It] should create and stop a working application
00:40:35   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:138
00:40:35 STEP: creating all guestbook components
00:40:35 INFO: Running '/jenkins-master-data/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/kubernetes/platforms/linux/amd64/kubectl kubectl --server=https://130.211.188.238 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/.kube/config create -f /jenkins-master-data/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/kubernetes/examples/guestbook --namespace=e2e-tests-kubectl-xcdto'
00:40:35 INFO: replicationcontrollers/frontend
00:40:35 services/frontend
00:40:35 replicationcontrollers/redis-master
00:40:35 services/redis-master
00:40:35 replicationcontrollers/redis-slave
00:40:35 services/redis-slave
00:40:35 
00:40:35 STEP: validating guestbook app
00:40:35 INFO: Waiting for frontend to serve content.
00:41:38 INFO: Trying to add a new entry to the guestbook.
00:41:39 INFO: Verifying that added entry can be retrieved.
00:41:39 STEP: using stop to clean up resources
00:41:39 INFO: Running '/jenkins-master-data/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/kubernetes/platforms/linux/amd64/kubectl kubectl --server=https://130.211.188.238 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/.kube/config stop -f /jenkins-master-data/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/kubernetes/examples/guestbook --namespace=e2e-tests-kubectl-xcdto'
00:41:45 INFO: replicationcontrollers/frontend
00:41:45 services/frontend
00:41:45 replicationcontrollers/redis-master
00:41:45 services/redis-master
00:41:45 replicationcontrollers/redis-slave
00:41:45 services/redis-slave
00:41:45 
00:41:45 INFO: Running '/jenkins-master-data/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/kubernetes/platforms/linux/amd64/kubectl kubectl --server=https://130.211.188.238 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-upgrade-gke-step1-deploy/workspace/.kube/config get pods,rc,se -l name=frontend --no-headers --namespace=e2e-tests-kubectl-xcdto'
00:41:46 INFO: frontend-8pq4q   1/1       Running   0         1m
00:41:46 frontend-nf2ym   1/1       Running   0         1m
00:41:46 frontend-oqi0w   1/1       Running   0         1m
00:41:46 
00:41:46 [AfterEach] Kubectl client
00:41:46   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:77
00:41:46 STEP: Destroying namespace for this suite e2e-tests-kubectl-xcdto
00:41:46 
00:41:46 • Failure [80.847 seconds]
00:41:46 Kubectl client
00:41:46 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:241
00:41:46   Guestbook application
00:41:46   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:139
00:41:46     should create and stop a working application [It]
00:41:46     /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:138
00:41:46 
00:41:46     Resources left running after stop:
00:41:46     frontend-8pq4q   1/1       Running   0         1m
00:41:46     frontend-nf2ym   1/1       Running   0         1m
00:41:46     frontend-oqi0w   1/1       Running   0         1m
00:41:46 
00:41:46     /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/util.go:759
```

**Comment by ihmccreery** [comment]
As far as I know, this was resolved (as #15123).
